src/activities/GenerateQueryWhereClause.ts

```diff
@@ -62,7 +62,12 @@ export default class GenerateWhereClause implements IActivityHandler {
         let where = whereClause && whereClause.length > 0 ? whereClause : "";
         const definitionExpression = getDefinitionExpression(layer);
         if (definitionExpression) {
-            where = `${definitionExpression} AND ${where}`;
+            if(where.length > 0) {
+                where = `${where} AND (${definitionExpression})`;
```

Comment:
You need parentheses around both parts. Consider the case where:
- `where` is `Field1 = 1 OR Field2 = 2` 
- `definitionExpression` is `Field3 = 3`

This would currently result in
`Field1 = 1 OR Field2 = 2 AND (Field3 = 3)`
which (due to operator precedence) is equivalent to 
`Field1 = 1 OR (Field2 = 2 AND (Field3 = 3))`
which isn't what we want.

The same is required on line 135.